ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -656,33 +656,205 @@ let text_document_lens (state : State.t)
+      let structure (iterator : Ast_iterator.iterator) structure =
+        List.iter structure ~f:(fun structure_item ->
+            iterator.structure_item iterator structure_item)
+      in
+      let signature (iterator : Ast_iterator.iterator) signature =
+        List.iter signature ~f:(fun signature_item ->
+            iterator.signature_item iterator signature_item)
+      in
+      let type_declaration (_iterator : Ast_iterator.iterator)
+          (type_declaration : Parsetree.type_declaration) =
+        let range = Range.of_loc type_declaration.ptype_loc in
+        push range
+      in
+      let module_type_declaration (iterator : Ast_iterator.iterator)
+          (module_type_declaration : Parsetree.module_type_declaration) =
+        let range = Range.of_loc module_type_declaration.pmtd_loc in
+        push range;
+        match module_type_declaration.pmtd_type with
+        | None -> ()
+        | Some module_type -> iterator.module_type iterator module_type
+      in
+      let module_type (iterator : Ast_iterator.iterator)
+          (module_type : Parsetree.module_type) =
+        match module_type.pmty_desc with
+        | Parsetree.Pmty_ident _ -> ()
+        | Parsetree.Pmty_signature signature ->
+          iterator.signature iterator signature
+        | Parsetree.Pmty_functor (_, _) -> ()
+        | Parsetree.Pmty_with (_, _) -> ()
+        | Parsetree.Pmty_typeof _ -> ()
+        | Parsetree.Pmty_extension _ -> ()
+        | Parsetree.Pmty_alias _ -> ()
+      in
+      let module_declaration (iterator : Ast_iterator.iterator)
+          (module_declaration : Parsetree.module_declaration) =
+        let range = Range.of_loc module_declaration.pmd_loc in
+        push range;
+        iterator.module_type iterator module_declaration.pmd_type
+      in
+      let class_declaration (_iterator : Ast_iterator.iterator)
+          (class_declaration : Parsetree.class_declaration) =
+        let range = Range.of_loc class_declaration.pci_loc in
+        push range
+      in
+      let value_binding (iterator : Ast_iterator.iterator)
+          (value_binding : Parsetree.value_binding) =
+        let range = Range.of_loc value_binding.pvb_loc in
+        push range;
+        iterator.expr iterator value_binding.pvb_expr
+      in
+      let expr (iterator : Ast_iterator.iterator) (expr : Parsetree.expression)
+          =
+        match expr.pexp_desc with
+        | Parsetree.Pexp_let (_, value_bindings, expr) ->
+          List.iter value_bindings ~f:(fun value_binding ->
+              iterator.value_binding iterator value_binding);
+          iterator.expr iterator expr
+        | Parsetree.Pexp_fun (_, _, _, expr) -> iterator.expr iterator expr
+        | Parsetree.Pexp_open (_, expr) -> iterator.expr iterator expr
+        | Parsetree.Pexp_letop letop ->
+          (* Location is not correct. It include the location of the whole
+             expression. See: https://github.com/ocaml/ocaml/pull/10682 *)
```

Comment:
Location is wrong (see link) with `letop` (let+,..). That's a bit annoying because when folding, it hides the whole expression. It works fine with regular `let` or `let%..`. Should it be better to disable support for `letop` for now ?

https://user-images.githubusercontent.com/5595092/147131513-6de7f6e1-4fdc-4ce4-a124-479b9632dfcb.mov